### PR TITLE
Draft updates

### PR DESCRIPTION
**Pull request by dekkerglen**
This fixes the issues that blocked the previous release.

### CODE REVIEW

File: one_shot_scripts/update_deck_format.js [hunk 1]
```diff
@@ -122,118 +121,121 @@ async function buildDeck(cards, bot) {
...
+      const bot = {
+        bot: deck.bots[i - 1],
+        pickorder: deck.cards[i].map((id) => {
+          if (typeof id === 'string' || id instanceof String) {
+            const details = carddb.cardFromId(id);
+            return {
+              tags: [],
+              colors: details.colors,
+              cardID: details._id,
+              cmc: details.cmc,
```
Comment (ruler501) [inline one_shot_scripts/update_deck_format.js]: We want to make sure this is defined if we haven't merged #1170 yet.

File: one_shot_scripts/update_deck_format.js [hunk 2]
```diff
@@ -122,118 +121,121 @@ async function buildDeck(cards, bot) {
...
+              type_line: details.type,
+            };
+          } else {
+            return id;
+          }
+        }),
+        name: 'Bot ' + (i + 1) + ': ' + deck.bots[i - 1][0] + ', ' + deck.bots[i - 1][1],
```
Comment (ruler501) [inline one_shot_scripts/update_deck_format.js]: Could we do `deck.bots[i - 1].join(', ')`?

File: one_shot_scripts/update_deck_format.js [hunk 3]
```diff
@@ -122,118 +121,121 @@ async function buildDeck(cards, bot) {
...
+        sideboard: botdeck.sideboard,
       };
+      deck.seats.push(bot);
+    }
+  } else {
+    //new format
+    deck.seats = [];
 
-      deck.seats.push(playerSeat);
+    //console.log(draft);
```
Comment (ruler501) [inline one_shot_scripts/update_deck_format.js]: Delete this line?

File: one_shot_scripts/update_deck_format.js [hunk 4]
```diff
@@ -122,118 +121,121 @@ async function buildDeck(cards, bot) {
...
+      const bot = {
+        bot: deck.bots[i],
+        pickorder: deck.cards[i].map((id) => {
+          if (typeof id === 'string' || id instanceof String) {
+            const details = carddb.cardFromId(id);
+            return {
+              tags: [],
+              colors: details.colors,
+              cardID: details._id,
+              cmc: details.cmc,
```
Comment (ruler501) [inline one_shot_scripts/update_deck_format.js]: Same comment about cmc.

File: one_shot_scripts/update_draft_format.js [hunk 5]
```diff
@@ -5,82 +5,127 @@ var carddb = require('../serverjs/cards.js');
...
+  if (Array.isArray(card)) {
+    return card.map(convertCard);
+  }
+  if (typeof card === 'string' || card instanceof String) {
+    const details = carddb.cardFromId(card);
+    return {
+      tags: [],
+      colors: details.colors,
+      cardID: details._id,
+      cmc: details.cmc,
```
Comment (ruler501) [inline one_shot_scripts/update_draft_format.js]: Same cmc comment.

File: one_shot_scripts/update_draft_format.js [hunk 6]
```diff
@@ -5,82 +5,127 @@ var carddb = require('../serverjs/cards.js');
...
-    userid: draft.owner,
-    name: draft.username,
-    pickorder: draft.pickOrder,
-    drafted: draft.picks[0],
-    packbacklog: draft.packs[0] && draft.packs[0][0] ? [draft.packs[0][0]] : [],
-  };
+async function update(draft) {
+  try {
+    if (draft.seats && draft.seats.length > 0) {
+      return async () => {};
```
Comment (ruler501) [inline one_shot_scripts/update_draft_format.js]: I don't think this needs to be async.

File: one_shot_scripts/update_draft_format.js [hunk 7]
```diff
@@ -5,82 +5,127 @@ var carddb = require('../serverjs/cards.js');
...
-          cardId: cardid,
-          details: carddb.cardFromId(cardid),
-        };
-        const col = Math.min(7, card.details.cmc) + (card.details.type.toLowerCase().includes('creature') ? 0 : 8);
-        bot.drafted[col].push(card);
+    //add bots
+    for (let i = 1; i < draft.picks.length; i++) {
+      const bot = {
+        bot: draft.bots[i - 1],
+        name: 'Bot ' + i + ': ' + draft.bots[i - 1][0] + ', ' + draft.bots[i - 1][1],
```
Comment (ruler501) [inline one_shot_scripts/update_draft_format.js]: Same comment about join.

File: one_shot_scripts/update_draft_format.js [hunk 8]
```diff
@@ -5,82 +5,127 @@ var carddb = require('../serverjs/cards.js');
...
+            cardid = cardid[0].cardID;
+          } else if (cardid.cardID) {
+            cardid = cardid.cardID;
+          }
+          //insert basic card object into correct cmc column
+          const card = {
+            cardId: cardid,
+            details: carddb.cardFromId(cardid),
+          };
+          const col = Math.min(7, card.details.cmc) + (card.details.type.toLowerCase().includes('creature') ? 0 : 8);
```
Comment (ruler501) [inline one_shot_scripts/update_draft_format.js]: Cmc might not be defined.

File: one_shot_scripts/update_draft_format.js [hunk 9]
```diff
@@ -5,82 +5,127 @@ var carddb = require('../serverjs/cards.js');
...
 
 (async () => {
   await carddb.initializeCardDb();
 
-  var i = 0;
   mongoose.connect(mongosecrets.connectionString).then(async (db) => {
-    const count = await Draft.countDocuments();
-    const cursor = Draft.find().cursor();
+    console.log(`starting...`);
+    const count = await Draft.estimatedDocumentCount();
```
Comment (ruler501) [inline one_shot_scripts/update_draft_format.js]: Don't we need an exact count here?

File: routes/cube_routes.js [hunk 10]
```diff
@@ -73,6 +74,323 @@ const { ensureAuth, csrfProtection, flashValidationErrors, jsonValidationErrors
...
+  'Marsh Flats': ['W', 'B'],
+  'Misty Rainforest': ['U', 'G'],
+  'Polluted Delta': ['U', 'B'],
+  'Scalding Tarn': ['U', 'R'],
+  'Verdant Catacombs': ['B', 'G'],
+  'Windswept Heath': ['W', 'G'],
+  'Wooded Foothills': ['R', 'G'],
+};
+
+function arrayIsSubset(needles, haystack) {
```
Comment (ruler501) [inline routes/cube_routes.js]: This is in dist/utils/Utils.

File: routes/cube_routes.js [hunk 11]
```diff
@@ -73,6 +74,323 @@ const { ensureAuth, csrfProtection, flashValidationErrors, jsonValidationErrors
 
 router.use(csrfProtection);
 
+const fetchLands = {
```
Comment (ruler501) [inline routes/cube_routes.js]: Can we put the bot rating code into a helper module like draftutil? These functions seem like they would fit well with the draft creation code there.

File: routes/cube_routes.js [hunk 12]
```diff
@@ -73,6 +74,323 @@ const { ensureAuth, csrfProtection, flashValidationErrors, jsonValidationErrors
...
+    };
+
+    draft.seats.push(playerSeat);
+    draft.unopenedPacks.push(draft.packs[0] ? draft.packs[0].slice(1) : []);
+
+    // add bots
+    for (let i = 1; i < draft.picks.length; i++) {
+      const bot = {
+        bot: draft.bots[i - 1],
+        name: `Bot ${i}: ${draft.bots[i - 1][0]}, ${draft.bots[i - 1][1]}`,
```
Comment (ruler501) [inline routes/cube_routes.js]: Join for colors please.

File: routes/cube_routes.js [hunk 13]
```diff
@@ -73,6 +74,323 @@ const { ensureAuth, csrfProtection, flashValidationErrors, jsonValidationErrors
...
+        drafted: [],
+        packbacklog: draft.packs[i] && draft.packs[i][0] ? [draft.packs[i][0]] : [],
+      };
+
+      // now we need to build picks from the pickorder ids
+      for (let j = 0; j < 16; j++) {
+        bot.drafted.push([]);
+      }
+
+      bot.pickorder.forEach((cardid, index) => {
```
Comment (ruler501) [inline routes/cube_routes.js]: It doesn't look like you need index so could this be done as a `for... of` loop?

File: routes/cube_routes.js [hunk 14]
```diff
@@ -73,6 +74,323 @@ const { ensureAuth, csrfProtection, flashValidationErrors, jsonValidationErrors
...
+      });
+
+      draft.seats.push(bot);
+      draft.unopenedPacks.push(draft.packs[i] ? draft.packs[i].slice(1) : []);
+    }
+    return draft;
+  } catch (err) {
+    req.logger.error(null, { error: err });
+  }
+  return async () => {};
```
Comment (ruler501) [inline routes/cube_routes.js]: Why return a callback?

File: routes/cube_routes.js [hunk 15]
```diff
@@ -73,6 +74,323 @@ const { ensureAuth, csrfProtection, flashValidationErrors, jsonValidationErrors
...
+              tags: [],
+              colors: details.colors,
+              cardID: details._id,
+              cmc: details.cmc,
+              type_line: details.type,
+            };
+          }
+          return id;
+        }),
+        name: `Bot ${i}: ${deck.bots[i][0]}, ${deck.bots[i][1]}`,
```
Comment (ruler501) [inline routes/cube_routes.js]: Join for colors please.

File: routes/root.js [hunk 16]
```diff
@@ -25,6 +27,272 @@ const router = express.Router();
 
 router.use(csrfProtection);
 
+//temprorary draft and deck conversion functions
```
Comment (ruler501) [inline routes/root.js]: Couldn't this be imported in instead of duplicated?

File: routes/root.js [hunk 17]
```diff
@@ -25,6 +27,321 @@ const router = express.Router();
 
 router.use(csrfProtection);
 
+const fetchLands = {
```
Comment (ruler501) [inline routes/root.js]: Same comment about putting in a helper.

File: routes/root.js [hunk 18]
```diff
@@ -161,19 +481,21 @@ router.get('/dashboard', async (req, res) => {
...
+    })
       .sort({
         date: -1,
       })
+      .lean()
       .limit(13);
 
+    decks = await Promise.all(decks.map(async (deck) => updateDeck(deck)));
+
+    console.log(decks[0]);
```
Comment (ruler501) [inline routes/root.js]: Unnecessary debug log statement.

File: routes/users_routes.js [hunk 19]
```diff
@@ -21,6 +21,321 @@ const router = express.Router();
 
 const { ensureAuth, csrfProtection, flashValidationErrors } = require('./middleware');
 
+const fetchLands = {
```
Comment (ruler501) [inline routes/users_routes.js]: Same comment on putting in a helper file.

File: src/components/DeckPreview.js [hunk 20]
```diff
@@ -8,6 +8,8 @@ const DeckPreview = ({ deck }) => {
   const maxLength = 35;
   const { date } = deck;
 
+  console.log(deck);
```
Comment (ruler501) [inline src/components/DeckPreview.js]: Unnecessary debug log statement.

File: src/pages/CubeDeckPage.js [hunk 21]
```diff
@@ -31,6 +31,9 @@ const CubeDeckPage = ({ cube, deck, canEdit, userid, draft }) => {
     [isOpen],
   );
 
+  console.log(draft);
```
Comment (ruler501) [inline src/pages/CubeDeckPage.js]: Unnecessary debug log statements.

File: routes/root.js (hunk 17 above)
Comment (dekkerglen) [reply routes/root.js]: this is a temporary function for the database migration

File: routes/users_routes.js (hunk 19 above)
Comment (dekkerglen) [reply routes/users_routes.js]: this is a temporary function for the database migration

**Pull request merged**